Caltech-IPAC/firefly
# FIREFLY-1358: Convert Range Slider and Version Info to JoyUI

#### [Firefly-1358](https://jira.ipac.caltech.edu/browse/FIREFLY-1358)

**Updates 12/12**
- converted as much of `FitsRotationDialog` and `ColorTableDropDownView` to joy UI as possible. Tried to remain consistent in the changes (and the overall look of the UI as well) 
- The Joy UI slider accepts a different format of `marks` object (array or bool) and I see that previously we had a check for it in the `checkMarksObject` function in `RangeSliderView.jsx`. Is this still necessary? 
- VersionInfo is pretty much the same as before. Attempted to spend some time converting `VersionInfo.jsx` to use Joy's `Grid` component, but I struggled a bit with getting this right. 
  - If it's a good idea to use `Grid`, I can spend some more time tomorrow to get it right/working. Other option is to come back to this part of `VersionInfo.jsx` when we start using Grid more widely so I have a better idea of how to use it right. 

**Testing**: 
- https://fireflydev.ipac.caltech.edu/firefly-1358-range-slider-joyui/firefly
- `VersionInfo` testing: 
  - Click on the firefly logo to get the Firefly Version Info dialog, make sure it looks ok. 
- `FitsRotationDialog` testing: 
  - Search for an image, click on the fits rotation dialog from our tools, ensure it looks okay. Use the slider and the input box to update the values. 
- `ColorTableDropDownView` testing:
  - Search for a fits image, say m81. Test the color table dropdown, and make sure it looks okay. You should clearly see `Color Bar`, `Bias` and `Contrast`. 
  - Search for another image, but this time for image type, choose `Create 3-color Composite`. Search three different times selecting red, green and blue. 
    - Click on the color dropdown now, you should see `RED Bias, RED Contrast `, `GREEN Bias, GREEN Contrast`, `BLUE Bias, BLUE Contrast` clearly. Play around with the slider. 
  - **_Important_**: For both of these dropdowns, click on the little popout arrow (top right) to open the dropdown as a PopupPanel and ensure the UI still looks okay.
